src/tests/builders/publish.rs

```diff
@@ -133,33 +121,105 @@ impl PublishBuilder {
         self
     }
 
+    pub fn no_manifest(mut self) -> Self {
+        self.manifest = Manifest::None;
+        self
+    }
+
+    pub fn custom_manifest(mut self, manifest: impl Into<Bytes>) -> Self {
+        self.manifest = Manifest::Custom(manifest.into());
+        self
+    }
+
+    pub fn add_file(mut self, path: impl ToString, content: impl Into<Bytes>) -> Self {
+        self.files.push((path.to_string(), content.into()));
+        self
+    }
+
     pub fn build(self) -> (String, Vec<u8>) {
         let new_crate = u::EncodableCrateUpload {
             name: u::EncodableCrateName(self.krate_name.clone()),
-            vers: u::EncodableCrateVersion(self.version),
-            features: self.features,
-            deps: self.deps,
-            description: self.desc,
+            vers: u::EncodableCrateVersion(self.version.clone()),
+            features: self.features.clone(),
+            deps: self.deps.clone(),
+            description: self.desc.clone(),
```

Comment:
I'm not particularly happy about all of the cloning here, but I haven't found a better way yet without a significant code complexity increase...